lib/logstash/inputs/http.rb

```diff
@@ -142,10 +143,12 @@ def run(queue)
           event.set("headers", req)
           decorate(event)
           queue << event
+          metric.increment(:events)
```

Comment:
I still think we should remove this; We could wrap the queue sent to a plugin with an instrumented queue that would track theses metrics.

**Rought proposal**

```ruby
class InstrumentedQueue
  def initialize(queue, plugin)
    @queue = queue
    @metric = plugin.metric
  end

  def push(item)
     timer = @metric.timer(:wait_on_queue_ms)
     @queue << item
      timer.stop
     @metric.increment(:events_in)
  end
end
alias_method :<<, :push
```

I believe I have discussed it with @jordansissel